crates/bevy_core_pipeline/src/core_3d/mod.rs

```diff
@@ -176,14 +183,16 @@ impl CachedRenderPipelinePhaseItem for Opaque3d {
 
 pub struct AlphaMask3d {
     pub distance: f32,
+    pub per_object_binding_dynamic_offset: u32,
```

Comment:
this is the dynamic binding offset, but in this context it's a "batch id". i think naming it that would make this code more comprehensible, but i haven't really thought it through / don't feel strongly about it.

Reply:
Later I will add more dynamic offsets into 'phase items', the idea being that a 'phase item' should include basically everything needed to identify whether two such items can be drawn together in one draw command or not. That includes pipeline, bind groups and dynamic offsets of the buffers they contain, index and vertex buffers, offsets and counts of things and so on. Or 'proxies' thereof - some kind of mesh id (currently a Handle<Mesh> or its id) would be analogous to the mesh's index and vertex buffers, and offsets and counts of indices or so.

Reply:
I'll add a comment to clarify though.